frontend/src/pages/integrations/railway/create.tsx

```diff
@@ -37,37 +50,39 @@ export default function RailwayCreateIntegrationPage() {
   const { data: integrationAuthApps } = useGetIntegrationAuthApps({
     integrationAuthId: (integrationAuthId as string) ?? ""
   });
+
   const { data: targetEnvironments } = useGetIntegrationAuthRailwayEnvironments({
     integrationAuthId: (integrationAuthId as string) ?? "",
-    appId: targetAppId
+    appId: watch("targetAppId")
```

Comment:
It would be better to keep a single watch at top and then use it everywhere. 

`const appId = watch("targetAppId")` => Then use this below, instead of registering multiple yet same proxy